alacritty_terminal/src/input.rs

```diff
@@ -996,6 +986,72 @@ mod tests {
 
     const KEY: VirtualKeyCode = VirtualKeyCode::Key0;
 
+    type MockBinding = Binding<usize>;
+
+    impl Default for MockBinding {
+        fn default() -> Self {
+            Self {
+                mods: Default::default(),
+                action: Default::default(),
+                mode: Default::default(),
+                notmode: Default::default(),
+                trigger: Default::default(),
+            }
+        }
+    }
+
+    #[test]
+    fn binding_matches_itself() {
+        let binding = MockBinding::default();
+        let identical_binding = MockBinding::default();
+
+        assert!(binding.triggers_match(&identical_binding));
+        assert!(identical_binding.triggers_match(&binding));
+    }
+
+    #[test]
+    fn binding_matches_different_action() {
+        let binding = MockBinding::default();
+        let mut different_action = MockBinding::default();
+        different_action.action = Action::ClearHistory;
+
+        assert!(binding.triggers_match(&different_action));
+        assert!(different_action.triggers_match(&binding));
+    }
+
+    #[test]
+    fn subset_mode_binding_matches_superset() {
+        let mut superset_mode = MockBinding::default();
+        superset_mode.mode = TermMode::ALT_SCREEN | TermMode::INSERT | TermMode::ORIGIN;
+        let mut subset_mode = MockBinding::default();
+        subset_mode.mode = TermMode::ALT_SCREEN | TermMode::ORIGIN;
+
+        assert!(superset_mode.triggers_match(&subset_mode));
+        assert!(!subset_mode.triggers_match(&superset_mode));
+    }
+
+    #[test]
+    fn subset_notmode_binding_matches_superset() {
+        let mut superset_notmode = MockBinding::default();
+        superset_notmode.notmode = TermMode::ALT_SCREEN | TermMode::INSERT | TermMode::ORIGIN;
+        let mut subset_notmode = MockBinding::default();
+        subset_notmode.notmode = TermMode::ALT_SCREEN | TermMode::ORIGIN;
+
+        assert!(superset_notmode.triggers_match(&subset_notmode));
+        assert!(!subset_notmode.triggers_match(&superset_notmode));
+    }
+
+    #[test]
+    fn mods_binding_requires_strict_match() {
+        let mut superset_mods = MockBinding::default();
+        superset_mods.mods = ModifiersState { alt: true, logo: true, ctrl: true, shift: true };
+        let mut subset_mods = MockBinding::default();
+        subset_mods.mods = ModifiersState { alt: true, logo: false, ctrl: false, shift: false };
+
+        assert!(!superset_mods.triggers_match(&subset_mods));
+        assert!(!subset_mods.triggers_match(&superset_mods));
+    }
+
```

Comment:
Might be nice to add a test for `is_triggered_by` to cover the change in functionality.